mage_ai/authentication/providers/sso.py

```diff
@@ -0,0 +1,20 @@
+from abc import ABC, abstractmethod
+from typing import Dict
+
+
+class SsoProvider(ABC):
+    """
+    Base class for single sign on providers. Can be used in conjunction with OauthProvider.
+    """
+    @abstractmethod
+    async def get_user_info(self, **kwargs) -> Dict:
+        """
+        This method should return a dictionary containing the user's information.
+
+        Returns:
+            Dict: dictionary with the following parameters
+                email (str): user's email
+                username (str): username
+                user_roles (List[Role], optional): list of user roles for the user
```

Comment:
Is this a TODO?
The GoogleProvider and OktaProvider don't return `user_roles` now.
Also, I notice you use `roles = user_info.get('roles')` in the SessionResource. Make sure the key name aligns.

Reply:
yeah it's a TODO. okta can return user role/permission info, but i just haven't implemented it yet.